miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppStorage.kt

```diff
@@ -1,4 +1,35 @@
 package com.rakuten.tech.mobile.miniapp.storage
 
+import com.rakuten.tech.mobile.miniapp.legacy.core.utils.LocalUrlParser
+import okhttp3.ResponseBody
+import java.io.File
+
+private const val SUB_DIR_MINIAPP = "miniapp"
+
 @SuppressWarnings("UseDataClass")
-internal class MiniAppStorage(val fileWriter: FileWriter)
+internal class MiniAppStorage(
+    val fileWriter: FileWriter,
+    val basePath: File,
+    val localUrlParser: LocalUrlParser = LocalUrlParser()
+) {
+
+    suspend fun saveFile(
+        response: ResponseBody,
+        basePath: String,
+        filePath: String,
+        fileName: String
+    ) = fileWriter.write(response, getAbsoluteWritePath(basePath, filePath, fileName))
+
+    fun getAbsoluteWritePath(
+        basePath: String,
+        filePath: String,
+        fileName: String
+    ) = "$basePath$filePath$fileName"
+
+    fun getFilePathFromUrl(file: String) = localUrlParser.getFilePath(file)
```

Comment:
Maybe `getFilePathFromUrl` and `getFileName` should be part of `MiniAppDownloader`? It looks like that's the only place that they're used. Also, these functions are specific to parsing the path from the URL, right? So it seems like it's more related to downloading than storage.

Also, that can remove our dependency on the legacy `LocalUrlParser`.

Reply:
It's a kind of one-leg-here-one-leg-there, the Storage class is interfacing in a way between the `FileWriter` and the `MiniAppDownloader`. 
The downloader has no intelligence right now on how to save the file, or where to, or how to parse the information related to storing of the file from the URL, all it knows is how to get a response stream against a given URL (or maybe in future about parallel downloads). The storage is in a way doing that for downloader.

Reply:
Yeah, it can be difficult to distinguish what's part of downloading and what's part of storage.

But I still think this functionality is more related to Downloading. I think of the relative file path as being a parameter of the response (in a way, it kind of is because it's encoded in the URL). So right now the storage has to know something about the API (that the file path is encoded in the URL), so this relative path parameter should instead be provided to the storage.

I'm just thinking about if we change how we're retrieving files, i.e. we retrieve a .zip file instead of individual files. We would probably have a separate class for unzipping files, but there would no longer be a URL to parse the relative file path from.

Reply:
I have noted it. I think it is calling in for an Adapter maybe (as it might just serve the exact purpose in this situation).